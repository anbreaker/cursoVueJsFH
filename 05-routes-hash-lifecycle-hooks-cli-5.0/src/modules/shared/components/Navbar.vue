<template>
  <div>
    <!-- <router-link :to="{ name: 'home' }">Pokemon List</router-link>
    <router-link :to="{ name: 'pokemon-id', params: { id: 81 } }">
      Pokemon by ID
    </router-link>
    <router-link :to="{ name: 'about' }">Pokemon About</router-link> -->
    <CustomLink v-for="link in links" :key="link.to" :link="link" />
  </div>
</template>

<script>
  import { defineAsyncComponent } from 'vue';

  export default {
    data() {
      return {
        links: [
          { to: '/home', name: 'Pokemons' },
          { to: '/pokemonid/50', name: 'Pokemons By Id' },
          { to: '/about', name: 'PokemonsById' },
          { to: 'https://google.com', name: 'google' },
        ],
      };
    },
    components: {
      CustomLink: defineAsyncComponent(() =>
        import(/* webpackChunkName: "CustomLink" */ './CustomLink')
      ),
    },
  };
</script>

<style scoped>
  div {
    padding: 0 30px;
    margin-bottom: 2rem;
  }

  div a {
    font-weight: bold;
    margin: 0 1rem;
    color: #fff;
    text-decoration: none;
  }

  /* div .router-link-exact-active {
    color: #42b983;
  } */
</style>
