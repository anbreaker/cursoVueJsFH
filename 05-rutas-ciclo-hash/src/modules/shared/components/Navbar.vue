<template>
  <div>
    <!-- <router-link :to="{ name: 'listPage' }">Pokemon List</router-link>
    <router-link :to="{ name: 'pokemon-id', params: { id: 85 } }">
      Pokemon por id
    </router-link>
    <router-link :to="{ name: 'aboutPage' }">About</router-link> -->
    <div class="inline">
      <CustomLink v-for="link in links" :key="link.to" :link="link" />
    </div>
  </div>
</template>

<script>
import { defineAsyncComponent } from "vue";

export default {
  data() {
    return {
      links: [
        { to: "pokemon-home", name: "pokemons" },
        { to: "pokemon-id", name: "Id", id: 151 },
        { to: "pokemon-about", name: "About" },

        { to: "dbz-characters", name: "dbz-characters" },
        { to: "dbz-about", name: "dbz-about" },

        { to: "https://google.com", name: "Google" }
      ]
    };
  },
  components: {
    CustomLink: defineAsyncComponent(() =>
      import(/* webpackChunkName: "CustomLink" */ "./CustomLink.vue")
    )
  }
};
</script>

<style scoped>
div {
  margin-top: 2rem;
  margin-bottom: 2rem;
  padding: 0 30px;
}

div a {
  font-weight: bold;
  color: aliceblue;
  text-decoration: none;
  font-size: 1.2rem;
  margin: 0 1rem;
}

a.router-link-exact-active {
  color: #42b893;
}

.inline {
  display: flex;
  justify-content: center;
}
</style>