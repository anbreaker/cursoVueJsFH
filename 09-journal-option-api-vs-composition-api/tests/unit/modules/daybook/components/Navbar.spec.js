import { shallowMount } from '@vue/test-utils';

import Navbar from '@/modules/daybook/components/Navbar.vue';
import { createVuexStore } from '../../../../../tests/mock-data/mock-store';

describe('Navbar Component tests', () => {
  const store = createVuexStore({
    user: {
      name: 'anbreaker',
      email: 'test@test.com',
    },

    status: 'authenticated',
    idToken: 'abc',
    refreshToken: 'xyz',
  });

  test('must display the component correctly', () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should be close logout', async () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    await wrapper.find('button').trigger('click');

    // TODO to be continued...
    expect(wrapper.router.push).toHaveBeenCalledWith();

    expect(store.state.auth).toEqual({
      idToken: null,
      refreshToken: null,
      status: 'not-authenticated', // 'authenticated', 'not-authenticated', 'authenticating'
      user: null,
    });
  });
});
