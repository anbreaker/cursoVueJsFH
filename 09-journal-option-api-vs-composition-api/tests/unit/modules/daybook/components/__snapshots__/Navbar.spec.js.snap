// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component tests must display the component correctly 1`] = `
<nav class="navbar bg-primary"><a class="navbar-brand text-white"><img src="" height="24" alt="Vue Logo" class="d-inline align-text-top mx-2"> anbreaker</a>
  <div class="d-flex"><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i> Logout </button></div>
</nav>
`;
